File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/Inventory/Sales.cs
﻿using LMS_CMS_DAL.Models.Domains.AccountingModule;
using LMS_CMS_DAL.Models.Domains.LMS;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.Inventory
{
    public class Sales : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name cannot be longer than 100 characters.")]
        public string Name { get; set; }
        public int InvoiceNumber { get; set; }
        public string Date { get; set; }
        public bool IsCash { get; set; }
        public bool IsVisa { get; set; }
        public int Amount { get; set; }
        public int Remaining { get; set; }
        public string Notes { get; set; }

        [ForeignKey("Store")]
        public long StoreID { get; set; }
        
        [ForeignKey("Student")]
        public long StudentID { get; set; }
        
        [ForeignKey("Save")]
        public long? SaveID { get; set; }
        
        [ForeignKey("Bank")]
        public long? BankID { get; set; }

        public Store Store { get; set; }
        public Student Student { get; set; }
        public Save? Save { get; set; }
        public Bank? Bank { get; set; }

        public ICollection<SalesItem> SalesItem { get; set; } = new HashSet<SalesItem>();
    }
}

File: LMS_CMS_Frontend/src/app/Models/Inventory/sales.ts
import { SalesItem } from "./sales-item";

export class Sales {
     constructor(
            public id: number = 0,
            public name: string = '',
            public invoiceNumber :number =0,
            public date: string = '',
            public isCash: boolean = false,
            public isVisa: boolean = false,
            public cashAmount :number =0,
            public visaAmount :number =0,
            public remaining :number =0,
            public total :number =0,
            public notes :string = '',
            public flagId :number =0,
            public storeID :number =0,
            public studentID :number =0,
            public saveID :number =0,
            public bankID :number =0,
            public storeName :string = '',
            public studentName :string = '',
            public saveName :string = '',
            public bankName :string = '',
            public insertedAt :string ="",
            public insertedByUserId :number =0,
            public attachments : string [] =[],
            public attachment : File [] =[] ,
            public NewAttachments : File [] =[] ,
            public DeletedAttachments : string [] =[] ,
            public inventoryDetails : SalesItem [] =[] ,
        ) {}
}

